cmd/genji/main.go

```diff
@@ -33,26 +33,35 @@ func main() {
 			return cli.NewExitError("cannot use bolt and badger options at the same time", 2)
 		}
 
-		dbpath := c.Args().First()
+		var opts *shell.Options
+		var err error
 
-		if (useBolt || useBadger) && dbpath == "" {
-			return cli.NewExitError("db path required when using bolt or badger", 2)
+		// check if there is a dbpath or a command.
+		if len(c.Args()) >= 1 {
+			opts, err = shell.GetOpts(c.Args())
+			if err != nil {
+				fmt.Println(err)
```

Comment:
Except for some very specific reasons, we should never use os.Exit in functions, we should return a `cli.NewExitError` here

Reply:
Using the CLI framework to parse has simplified this part.
We really need to use `log.Fatal(err)` in the `main` which display a datetime:
```
if err := app.Run(os.Args); err != nil {
		log.Fatal(err)
	}

# The output 
2020/09/09 00:05:48 table not found
```

`fmt.Println(err)` doesn't display a datetime.